File: frontend/src/types/task.ts
export type TaskKanbanColumnId = 'overdue' | 'today' | 'tomorrow' | 'later' | 'no-deadline'

export type TaskStatus = 'active' | 'completed'

export type TaskActivityType = 'system' | 'comment'

export interface TaskActivityEntry {
  id: string
  type: TaskActivityType
  author: string
  text: string
  createdAt: Date
}

export interface TaskAttachment {
  id: string
  name: string
  size: number
  mimeType: string
  uploadedBy: string
  uploadedAt: Date
}

export interface Task {
  id: string
  title: string
  text: string
  dueAt: number | null
  leadId?: string
  dealId?: string
  createdBy: string
  createdAt: Date
  status: TaskStatus
  activities: TaskActivityEntry[]
  attachments: TaskAttachment[]
}

export interface NewTaskPayload {
  title: string
  text: string
  dueAt: number | null
  leadId?: string
  dealId?: string
  createdBy?: string
}

export interface UpdateTaskPayload {
  title?: string
  text?: string
  dueAt?: number | null
}

File: backend/internal/model/task.go
package model

type Task struct {
	ID               string       `json:"id"`
	Title            string       `json:"title"`
	Text             string       `json:"text"`
	DueAt            *int64       `json:"dueAt"`
	Status           string       `json:"status"`
	LeadID           *string      `json:"leadId,omitempty"`
	DealID           *string      `json:"dealId,omitempty"`
	ClientFirstName  string       `json:"clientFirstName"`
	ClientPatronymic string       `json:"clientPatronymic"`
	ClientPhone      string       `json:"clientPhone"`
	TrafficSource    string       `json:"trafficSource"`
	CreatedBy        string       `json:"createdBy"`
	CreatedAt        int64        `json:"createdAt"`
	UpdatedAt        int64        `json:"updatedAt"`
	Attachments      []Attachment `json:"attachments"`
	Activities       []Activity   `json:"activities"`
}

type CreateTaskInput struct {
	Title     string  `json:"title"`
	Text      string  `json:"text"`
	DueAt     *int64  `json:"dueAt"`
	Status    string  `json:"status"`
	LeadID    *string `json:"leadId"`
	DealID    *string `json:"dealId"`
	CreatedBy string  `json:"createdBy"`
}

type UpdateTaskInput struct {
	Title    *string `json:"title"`
	Text     *string `json:"text"`
	DueAt    *int64  `json:"dueAt"`
	HasDueAt bool
}
